client/verta/verta/client.py

```diff
@@ -1379,15 +1385,23 @@ class Experiment(_ModelDBEntity):
         Experiment Runs under this Experiment.
 
     """
-    def __init__(self, conn, conf,
+    def __init__(self, conn, conf, id_):
+        super(Experiment, self).__init__(conn, conf, _ExperimentService, "experiment", id_)
+
+        self._conn = conn
+
+        self.id = id_
+
+    @staticmethod  # TODO: if PR approved, move this fn down with other static/class methods
+    def _get_or_create(conn, conf,
```

Comment:
Here's where the logic is moved out of `Experiment.__init__()` and into `_get_or_create()`